classes/order/Order.php

```diff
@@ -1673,7 +1673,7 @@ public function getWsOrderRows()
      */
     public function setCurrentState($id_order_state, $id_employee = 0)
     {
-        if (empty($id_order_state)) {
+        if (empty($id_order_state) || $id_order_state == $this->current_state) {
```

Comment:
```suggestion
        if (empty($id_order_state) || (int) $id_order_state === (int) $this->current_state) {
```
Let's have a strict comparison :)